syncapi/sync/requestpool_test.go

```diff
@@ -125,11 +128,19 @@ func TestRequestPool_updatePresence(t *testing.T) {
 	go rp.cleanPresence(db, time.Millisecond*50)
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			beforeCount := publisher.count
+			beforeCount := func() int {
```

Comment:
Probably easier to `Lock`/`Unlock` before/after setting `beforeCount`, for readability.